src/rgw/rgw_op.h

```diff
@@ -1515,6 +1515,10 @@ class RGWCopyObj : public RGWOp {
 
   bool need_to_check_storage_class = false;
 
+  //object lock
+  RGWObjectRetention *obj_retention;
+  RGWObjectLegalHold *obj_legal_hold;
```

Comment:
these need to be initialized to `nullptr`, and deleted in the `RGWCopyObj` destructor